### PR TITLE
Remove any path in the compilation database parameters

### PR DESCRIPTION
**Pull request by Moeryn**
I had the following error when my compilation database was loaded:

```
[deoplete] Traceback (most recent call last):                                                                                                                                                                                                                                                                                  
[deoplete]   File "/home/pcaneill/.vim/bundle/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 98, in gather_results                                                                                                                                                                                                  
[deoplete]     ctx['candidates'] = source.gather_candidates(ctx)                                                                                                                                                                                                                                                               
[deoplete]   File "/home/pcaneill/.vim/bundle/deoplete-clang/rplugin/python3/deoplete/sources/deoplete_clang.py", line 151, in gather_candidates                                                                                                                                                                               
[deoplete]     params)                                                                                                                                                                                                                                                                                                         
[deoplete]   File "/home/pcaneill/.vim/bundle/deoplete-clang/rplugin/python3/deoplete/sources/deoplete_clang.py", line 314, in get_completion                                                                                                                                                                                  
[deoplete]     tu = self.get_translation_unit(fname, args, buf)                                                                                                                                                                                                                                                                
[deoplete]   File "/home/pcaneill/.vim/bundle/deoplete-clang/rplugin/python3/deoplete/sources/deoplete_clang.py", line 301, in get_translation_unit                                                                                                                                                                            
[deoplete]     tu = self.index.parse(fname, args, buf, flags)                                                                                                                                                                                                                                                                  
[deoplete]   File "/home/pcaneill/.vim/bundle/deoplete-clang/rplugin/python3/deoplete/clang/clang/cindex.py", line 2322, in parse                                                                                                                                                                                              
[deoplete]     self)                                                                                                                                                                                                                                                                                                           
[deoplete]   File "/home/pcaneill/.vim/bundle/deoplete-clang/rplugin/python3/deoplete/clang/clang/cindex.py", line 2440, in from_source                                                                                                                                                                                        
[deoplete]     raise TranslationUnitLoadError("Error parsing translation unit.")                                                                                                                                                                                                                                               
[deoplete] clang.cindex.TranslationUnitLoadError: Error parsing translation unit.                                                                                                                                                                                                                                              
[deoplete] Could not get completions from: clang.  Use :messages for error details. 
```

When I looked at the params given to the translation unit it was something like this:

```
 ['-I/dfqdfqdfqd', '-Wblublu', '/path/to/the/file']
```

This patch aims to remove /path/to/the/file

### CODE REVIEW

**Comment by Shougo** [comment]
Please support Windows path pattern.
You should use `os.path.isabs(path)`.

File: rplugin/python3/deoplete/sources/deoplete_clang.py [hunk 1]
```diff
@@ -265,7 +265,7 @@ def get_compilation_database(self, fname):
                              os.path.realpath(os.path.join(cwd, arg))]:
                         skip = 0
                         continue
-                    elif arg == '-o':
+                    elif arg == '-o' or arg[0] == '/':
```
Comment (Moeryn) [inline rplugin/python3/deoplete/sources/deoplete_clang.py]: Actually this is not the right fix.

The first if line 263 should remove the file name after the '-c', I don't understand why it did'nt work in my case.

How can I debug this? For now raised Exception that was catch by deoplete, but is there an other way?

Comment (Shougo) [inline rplugin/python3/deoplete/sources/deoplete_clang.py]: @Moeryn Please upload the `param` value.
And please check the wrong argument.

**Comment by parnmatt** [comment]
I too am having this issue, @Moeryn have you made any further progress on this?

**Comment by Shougo** [comment]
@parnmatt You can check `param` value.
The correct information is needed.

**Comment by parnmatt** [comment]
Unsure why, but the debug settings were not working; so I simply printed the arguments to a file.

The values of the `args` parameter that reached the translation unit (which are the `params` sent from this plugin) are: `['-x', 'c++', '-std=c++1z', '-I']`.

This is when writing in a simple test file, and the error is thrown after type `st`.

**Comment by Shougo** [comment]
Please upload the compilation database.
I will test it.

**Comment by Shougo** [comment]
I will close the issue.  Because, the original author @Moeryn is no response and it does not fix the real problem.

@parnmatt Please create new issue with minimal init.vim and the simple test file and the compilation database file.
I will test it.

**Pull request closed without merging**